### DISCUSSION TITLE
Search "within" a bookmark

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#13885] [Archive]
Hi guys,

One question, when I do a search within a collection and I get several pages of results I can consult the items, go back, and stay on the same page, which previously I always returned to the first one.

The same does not happen when we search within a bookmark, I do a search, I go to consult an item on page 5 ex. when i go back, it goes back to page 1..

Is it supposed to be this way or will it be put like in the collections?

**Another situation I saw was in Presets & Bookmarks, when I apply a filter I get a list if I select an item from that list and go back I lose the filter.. shouldn't it keep the filter?**

**Comment by rijkvanzanten** [comment]
> The same does not happen when we search within a bookmark, I do a search, I go to consult an item on page 5 ex. when i go back, it goes back to page 1..
>
> Is it supposed to be this way or will it be put like in the collections?

This is because you're technically going back to the bookmark, which also includes the current page you're on. The collection view saves any change to the current view, whereas bookmarks will have a dedicated button to save the changes to the bookmark 🤔 

> Another situation I saw was in Presets & Bookmarks, when I apply a filter I get a list if I select an item from that list and go back I lose the filter.. shouldn't it keep the filter?

That should've been resolved in https://github.com/directus/directus/pull/13821 👍🏻

**Reply by hugopvl** [reply]
> This is because you're technically going back to the bookmark, which also includes the current page you're on. The collection view saves any change to the current view, whereas bookmarks will have a dedicated button to save the changes to the bookmark 🤔
> 


Good Morning @rijkvanzanten 

I think I didn't explain myself well, or I'm missing something that I'm not doing well. A practical example I have several collections, one of them is footwear. within this category I have several bookmarks (shoes, sneekers, boots, slippers.. ) in any of the bookmarks I have more than 5000 records, which gives me a total of at least 5 pages, so I enter the shoes bookmark and start to see the items until I'm on page 3 I select an item to check the data, when I go back it goes back to page 1 and I have to reselect page 3 look for the record I consulted and continue the search. in this situation I thought I could go back to the previous position (page 3). otherwise every time I want to check an item before that I have to do update bookmark!?

**Reply by rijkvanzanten** [reply]
Yeah! I totally understood what you're saying 🙂 My message above goes into _why_ it currently works the way it does, not necessarily saying it's the expected/correct way 😁 

As of right now, the current "page" is stored as part of the preset/bookmark, which in turn means that it's reset when you leave-and-come-back to the bookmark, as bookmark changes aren't automatically saved. "Regular" content views do autosave any changes you make to the layout configuration, which is why the current `page` value is saved before navigating to the detail view.

The question here is what the expected behavior should be. If we'd auto-save changes to bookmarks as well, we'd solve for this problem, but you'd also lose the expected behavior of a bookmark, as it effectively just turns into another tab instead. If we were to auto-save only some of the properties, it might get confusing (for example, if we auto-save `page`, we also have to auto-save `sort`. What if you filter and then navigate to a detail view? In that case, `filter` should _also_ have been saved. At that point, we're just back at the problem mentioned earlier where we're just auto-saving all changes). 

Another potential solution would be to render the item detail view in a _drawer_ rather than a full new page (like we've brainstormed on ages ago @benhaynes 🙂) as that effectively negates the problem by leaving the layout in the background 🤔

**Reply by hugopvl** [reply]
> Another potential solution would be to render the item detail view in a _drawer_ rather than a full new page (like we've brainstormed on ages ago @benhaynes 🙂) as that effectively negates the problem by leaving the layout in the background 🤔

I think the idea of ​​the drawer for those who are consulting is cleaner and simpler ❤️
